common/types/mapper/thrift/shared.go

```diff
@@ -5474,60 +5474,80 @@ func ToTransientDecisionInfo(t *shared.TransientDecisionInfo) *types.TransientDe
 	}
 }
 
-// FromUpdateDomainInfo converts internal UpdateDomainInfo type to thrift
-func FromUpdateDomainInfo(t *types.UpdateDomainInfo) *shared.UpdateDomainInfo {
-	if t == nil {
-		return nil
-	}
-	return &shared.UpdateDomainInfo{
-		Description: t.Description,
-		OwnerEmail:  t.OwnerEmail,
-		Data:        t.Data,
-	}
-}
-
-// ToUpdateDomainInfo converts thrift UpdateDomainInfo type to internal
-func ToUpdateDomainInfo(t *shared.UpdateDomainInfo) *types.UpdateDomainInfo {
-	if t == nil {
-		return nil
-	}
-	return &types.UpdateDomainInfo{
-		Description: t.Description,
-		OwnerEmail:  t.OwnerEmail,
-		Data:        t.Data,
-	}
-}
-
 // FromUpdateDomainRequest converts internal UpdateDomainRequest type to thrift
 func FromUpdateDomainRequest(t *types.UpdateDomainRequest) *shared.UpdateDomainRequest {
 	if t == nil {
 		return nil
 	}
-	return &shared.UpdateDomainRequest{
-		Name:                     t.Name,
-		UpdatedInfo:              FromUpdateDomainInfo(t.UpdatedInfo),
-		Configuration:            FromDomainConfiguration(t.Configuration),
-		ReplicationConfiguration: FromDomainReplicationConfiguration(t.ReplicationConfiguration),
+	request := shared.UpdateDomainRequest{
+		Name:                     &t.Name,
 		SecurityToken:            t.SecurityToken,
 		DeleteBadBinary:          t.DeleteBadBinary,
 		FailoverTimeoutInSeconds: t.FailoverTimeoutInSeconds,
 	}
+	if t.Description != nil || t.OwnerEmail != nil || t.Data != nil {
+		request.UpdatedInfo = &shared.UpdateDomainInfo{
```

Comment:
I'm not entirely sure if this is a good idea, but an alternate tactic could be:
```go
request := shared.UpdateDomainRequest{
  // ...
  UpdatedInfo: &shared.UpdateDomainInfo{
    // ... all values
  }
  // same for others, just do it all in one big struct init
}

// erase subsections that are zero-valued
if *request.UpdatedInfo == shared.UpdateDomainInfo{} {
  request.UpdatedInfo = nil
}
```
it might be a little more resistant to forgetting a `!= nil` check, and seems like it might match the semantics a bit better ("zero value means nothing changed means nil" vs "these fields are nil means nil")

Reply:
This would be nicer, I agree. Unfortunately we can not compare these struct like that since they contain slices or maps.

Reply:
ah, right, un-comparable things.  fun.
well, there is `reflect.IsZero()`